### ISSUE TITLE
Rule Config Suggestion: Allow unused aliases for subquery

### ISSUE THREAD

**Issue by aaronsteers** [alanmcruickshank/sqlfluff#429]
Applies to rule **[L025](https://docs.sqlfluff.com/en/latest/rules.html#sqlfluff.rules.std.Rule_L025)** which is `Tables should not be aliased if that alias is not used.`

While I agree (and want to enable this rule) for normal tables, I found this also triggered when the "table" in question was a subquery.

Example which triggers the error:

```sql
SELECT
    col_a,
    sum(col_b) AS b_summed
FROM (
    SELECT
        table_a.col_a,
        table_b.col_b
    FROM table_a
    JOIN table_b
    USING (some_key)
) AS subquery
GROUP BY col_a
```

In the above example, the rule would impose either adding `subquery.` as a prefix on all the outer SELECT's column refs, or else removing the alias from its place after the subquery. I might be alone here, but I've always thought it would be an antipattern to create un-aliased subqueries - except of course, if that subquery is used inline as a scalar expression. (I'm happy to be wrong on this, just wanted to check.)

Thoughts?

**Comment by aaronsteers** [comment]
On further thought and exploration, I think omitting the subquery alias is actually plenty clean and still easily readable. Resolving as a non-issue.
